source/Octopus.Tentacle/Kubernetes/Diagnostics/KubernetesAgentMetrics.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Kubernetes.Diagnostics
+{
+    public class KubernetesAgentMetrics
+    {
+        public delegate KubernetesAgentMetrics Factory(IPersistenceProvider persistenceProvider);
+        
+        readonly IPersistenceProvider persistenceProvider;
+        readonly ISystemLog log;
+
+        public KubernetesAgentMetrics(IPersistenceProvider persistenceProvider, ISystemLog log)
+        {
+            this.persistenceProvider = persistenceProvider;
+            this.log = log;
+        }
+
+        public void TrackEvent(string reason, string source, DateTimeOffset occurrence)
+        {
+            try
+            {
+                lock (persistenceProvider)
+                {
+                    var existingEvent = LoadFromPersistence(reason) ?? new EventJsonEntry(source, new List<DateTimeOffset>());
```

Comment:
Maybe I'm not understanding this right but if two events come with the same reason but a different source come in, the second source will be lost?

Reply:
:thinking: this one could be a bug!